WebScripts~/src/WebRtcClient.ts

```diff
@@ -0,0 +1,172 @@
+            this.handleDc(id, dc);
+        } else {
+            pc.addEventListener("datachannel", (event) => this.handleDc(id, event.channel));
+        }
+    };
+
+    private handleDc = (id: string, dc: RTCDataChannel) => {
+        if (dc.label !== this.label) {
+            return;
+        }
+
+        if (this.isDebug) {
+            console.log(`New DataChannel: id=${id} label=${dc.label}`);
+        }
+
+        this.dcMap.set(id, dc);
+        this.idMapper.add(id);
+        const clientId = this.idMapper.get(id) as number;
+
+        // Host only
+        if (this.getPeerClient().role === PeerRole.Host) {
+            dc.addEventListener("open", () => {
+                if (this.isDebug) {
+                    console.log(`OnOpen: clientId=${clientId}`);
+                }
+                this.callbacks.onConnected(clientId);
+            });
+        }
+
+        // Both Host and Client
+        dc.addEventListener("message", (event) => {
+            this.callbacks.onDataReceived(clientId, event.data);
+        });
+        dc.addEventListener("close", () => {
+            if (this.isDebug) {
+                console.log(`OnClose: clientId=${clientId}`);
+            }
+
+            if (this.getPeerClient().role === PeerRole.Host && this.disconnectedRemoteClients.delete(clientId)) {
+                return;
+            }
+            this.callbacks.onDisconnected(clientId);
+        });
+    };
+
+    private closePc = (id: string) => {
+        const dc = this.dcMap.get(id);
+        if (!dc) {
+            return;
+        }
+        dc.close();
+        this.dcMap.delete(id);
+        this.idMapper.remove(id);
+    };
+
+    public connect = async () => {
+        if (this.isDebug) {
+            console.log(`Connect: role=${this.getPeerClient().role}`);
+        }
+
+        if (this.getPeerClient().role === PeerRole.Client) {
+            const hostId = this.getPeerClient().hostId;
+            if (hostId === null) {
+                return;
+            }
+            this.cancel = false;
+            await waitUntil(
+                () => this.idMapper.has(hostId),
+                () => this.cancel,
+            );
+            const clientId = this.getHostId("connect", hostId);
+            if (!this.isHostIdNotFound(clientId)) {
+                this.callbacks.onConnected(clientId);
+            }
+        }
+    };
+
+    private readonly hostIdNotFound = 0;
+    private isHostIdNotFound = (hostId: number) => hostId === this.hostIdNotFound;
+
+    private getHostId = (caller: string, hostId: string | null) => {
+        let result;
+        if (hostId !== null) {
+            result = this.idMapper.has(hostId) ? (this.idMapper.get(hostId) as number) : this.hostIdNotFound;
+        } else {
+            result = this.hostIdNotFound;
+        }
+        if (this.isDebug) {
+            console.log(`getHostId: caller=${caller} hostId=${hostId}`);
+        }
+        return result;
+    };
```

Comment:
```suggestion
    private getHostId = (caller: string, hostId: string | null) => {
        if (this.isDebug) {
            console.log(`getHostId: caller=${caller} hostId=${hostId}`);
        }
        return hostId !== null && this.idMapper.has(hostId) ? (this.idMapper.get(hostId) as number) : this.hostIdNotFound;
    };
```

ネイティブの方と同様でこちらの方がすっきりすると思います。

Reply:
fixed https://github.com/extreal-dev/Extreal.Integration.Multiplay.NGO/pull/21/commits/1c4b8c2560788aabac0c16e0fb27c7cb196a26e8

Reply:
確認しました！